collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2633,8 +2634,12 @@ public void onSaveFormIndexError(String errorMessage) {
     public void onNumberPickerValueSelected(int widgetId, int value) {
         if (currentView != null) {
             for (QuestionWidget qw : ((ODKView) currentView).getWidgets()) {
-                if (qw instanceof RangeWidget && widgetId == qw.getId()) {
-                    ((RangeWidget) qw).setNumberPickerValue(value);
+                if (qw instanceof RangePickerIntegerWidget && widgetId == qw.getId()) {
```

Comment:
I'm not a big fan of the `FormEntryActivity` having to know about widget implementations here but this structure existed a long time before this PR and I think it's be awkward to unravel.

Reply:
@seadowg I think the other option that we can use is having a separate `viewModel` for `RangeWidget`. I also don't like having a callback implemented in `FormEntryActivity`.